### PR TITLE
Translate to java, added observe

### PR DESCRIPTION
**Pull request by dengdan154**
Issue #, if available:

Description of changes:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: getting-started/todo/kotlin/app/.gitignore [hunk 1]
```diff
@@ -0,0 +1,21 @@
+/build
+
+#amplify
+amplify/\#current-cloud-backend
```
Comment (sdhuka) [inline getting-started/todo/kotlin/app/.gitignore:4]: use amplify/** instead

File: getting-started/todo/kotlin/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        android:allowBackup="true"
+        android:icon="@mipmap/ic_launcher"
+        android:label="@string/app_name"
+        android:roundIcon="@mipmap/ic_launcher_round"
+        android:supportsRtl="true"
+        android:theme="@style/Theme.Todo">
+        <activity android:name="com.amplifyframework.samples.gettingstarted.TodoListActivity">
+            <intent-filter>
+                <action android:name="android.intent.action.MAIN" />
+
```
Comment (sdhuka) [inline getting-started/todo/kotlin/app/src/main/AndroidManifest.xml:16]: np: remove blank line

File: getting-started/todo/kotlin/app/src/main/java/com/amplifyframework/samples/gettingstarted/TodoListActivity.kt [hunk 3]
```diff
@@ -30,9 +34,19 @@ class TodoListActivity : ListActivity(), TodoItemAdapter.OnItemClickListener {
...
+            itemAdapter.query(hideStatus)
+            swipeRefresh.isRefreshing = false
+        }
+
         val itemTouchHelper = ItemTouchHelper(swipeHandler)
         itemTouchHelper.attachToRecyclerView(recyclerView)
         ItemAdapter.setContext(this)
+
+        // Observe changes bi-directional
+        itemAdapter.observe()
```
Comment (sdhuka) [inline getting-started/todo/kotlin/app/src/main/java/com/amplifyframework/samples/gettingstarted/TodoListActivity.kt:49]: why do we need this here?

File: getting-started/todo/kotlin/app/src/main/res/layout/todo_item.xml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        android:text="@string/sample_text"
+        android:textSize="@dimen/text_size"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.0"
+        app:layout_constraintStart_toEndOf="@id/todo_checkbox"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="0.333" />
+
+    <LinearLayout
```
Comment (sdhuka) [inline getting-started/todo/kotlin/app/src/main/res/layout/todo_item.xml:32]: what is this used for?

Comment (dengdan154) [reply getting-started/todo/kotlin/app/src/main/res/layout/todo_item.xml:32]: Can you clarify what you are referring to?

**Review by sdhuka — APPROVED** [review]
LGTM!

**Pull request merged**